lib/Doctrine/Common/Reflection/TypedNoDefaultReflectionProperty.php

```diff
@@ -0,0 +1,25 @@
+<?php
+
+namespace Doctrine\Common\Reflection;
+
+use ReflectionProperty;
+
+/**
+ * PHP Typed No Default Reflection Property - special override for typed properties without a default value.
+ */
+class TypedNoDefaultReflectionProperty extends ReflectionProperty
+{
+    /**
+     * {@inheritDoc}
+     *
+     * Checks that a typed property is initialized before accessing its value.
+     * This is neccessary to avoid PHP error "Error: Typed property must not be accessed before initialization".
+     * Should be used only for reflecting typed properties without a default value.
+     */
+    public function getValue($object = null)
+    {
+        $name = $this->getName();
+
+        return isset($object->$name) ? parent::getValue($object) : null;
```

Comment:
I think an `isset($object->$name)` will initialize that property starting from ORM 3, because ProxyManager intercepts `isset()` calls via `__isset()`.

Instead, [`ReflectionProperty#isInitialized()`](https://3v4l.org/kFMpL) should be used.

Reply:
And yes: more performance impact :-(

Reply:
I can do this, but again it looks to me as a separate issue, this behavior was copied from another class in the package `\Doctrine\Common\Reflection\RuntimePublicReflectionProperty`, it has issues then too.
Thanks for the review!

Reply:
In general, accessing typed properties via `isset()` is not side-effect-free, and will break on anything declaring `__isset` (not just because of ProxyManager)

Reply:
Yes, I understand the problem now, will fix it in the new class, it will also make it's purpose more explicit. Thank you!

Reply:
Fixed https://github.com/doctrine/reflection/blob/f45415e985ee72c3b3559dae302975451caa5299/lib/Doctrine/Common/Reflection/TypedNoDefaultReflectionProperty.php#L19-L22